components/core/tests/test-ParserWithUserSchema.cpp

```diff
@@ -79,56 +93,55 @@ void decompress(std::string archive_dir, std::string output_dir) {
 }
 
 TEST_CASE("Test error for missing schema file", "[LALR1Parser][SchemaParser]") {
-    std::string file_path = "../tests/test_schema_files/missing_schema.txt";
-    std::string file_name = boost::filesystem::weakly_canonical(file_path).string();
+    auto const file_path = test_schema_dir / "missing_schema.txt";
     REQUIRE_THROWS_WITH(
-            generate_schema_ast(file_path),
-            "Failed to read '" + file_path + "', error_code="
+            generate_schema_ast(file_path.string()),
```

Comment:
sorry for my oversight. since we need to use file_path.string() twice for this test_case, let's store it as a separate variable
`auto const file_path_string = file_path.string()`